### DISCUSSION TITLE
Guard clauses

### DISCUSSION THREAD

**Discussion by jsellaresi** [jorsang1/clean-microservice-template#4] [General]
If we are going to use a _validation_ approach, the guard clauses should be removed from these classes.

https://github.com/jorsang1/clean-microservice-template/blob/b9942a930f647969d9e07878ac05d12b326c92b5/src/Domain/Entities/Product/ValueObjects/ProjectTitle.cs#L13

**Comment by jorsang1** [comment]
I agree with you.
I left this there to enable the discussion but IMO its better to validate everything the way we are doing it once everything is set, than 'guarding' in the constructor and throwing in every error it's found.
